### ISSUE TITLE
Vocabulary training dataset

### ISSUE THREAD

**Issue by machshev** [machshev/haqor-core#33]
Show verses based on the current vocabulary the reader is familiar with.  Introducing new words by selecting verses that make use of those words. 

For this to work there must be some way of recording how familiar a reader is with a given word. Ideally words are introduced by how common they are. Which opens up the possibility of showing more words. 

It might be possible to approximately identify how large someone's vocabulary is by assuming people learn words in frequency order. Then use a sort of gradient descent to determine a starting point. From there start tailoring for individuals based on feedback.

We could also differentiate between base vocabulary and grammatical forms. So while someone might know a word, they might not be able to see the root form. Verses could also be graded for grammatical complexity.

Initially it might be difficult to learn any verses. So maybe also have a flashcard system that shows words, then progresses to showing verses. Potentially even extend this idea to suggesting groups of verses or even chapters or books to read based on vocabulary.

Ideally an interface that works in the terminal would be good. But it might work better with a GUI.
